src/api.rs

```diff
@@ -80,20 +88,68 @@ impl API {
     };
   }
 
+  fn login() {
+    // Get credentials
+    let mut username = match env::var("CLINK_USERNAME") {
+      Ok(user) => {
+        if user.chars().all(char::is_alphanumeric) {
+          user
+        } else {
+          get_current_username().unwrap().into_string().unwrap()
+        }
+      }
+      Err(_) => get_current_username().unwrap().into_string().unwrap(),
+    };
+
+    println!(
+     "Please enter password for {}: ",
+     username
+    );
+    let password = read_password().unwrap();
+
+    // Start kinit, ready to get password from pipe
+    let process = Command::new("kinit")
+      .arg(format!("{}@CSH.RIT.EDU", username))
+      .stdin(Stdio::piped())
+      .stdout(Stdio::piped())
+      .spawn()
+      .unwrap();
```

Comment:
Does this just panic if it unwraps into something bad? The main concern I have here is, **does the user get appropriate feedback if their credentials are wrong?**

Reply:
Yes

<img width="763" alt="image" src="https://user-images.githubusercontent.com/40615740/157083194-75682224-6ef7-4aff-b69c-00aace7d5f9f.png">

First example is no password, 2nd is wrong password